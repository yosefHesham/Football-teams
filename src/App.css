* {
  margin: 0;
}

body {
  width: 100vw;
}

h1 {
  color: lightgrey;
}

.c-white {
  color: white;
}
a{
  text-decoration: none;
}
