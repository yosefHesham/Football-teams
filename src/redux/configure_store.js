import {applyMiddleware, createStore } from "redux";
import logger from "redux-logger";
import thunk from "redux-thunk";
import leaguesReducer from "./teams/teams_reducer";



const store = createStore(leaguesReducer,applyMiddleware(thunk,logger))

export default store;