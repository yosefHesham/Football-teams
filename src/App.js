import React from 'react';

import './App.css';
import RouteHandler from './components/routeHandler';

const App = () => (
  <main>
    <RouteHandler />
  </main>
);

export default App;
