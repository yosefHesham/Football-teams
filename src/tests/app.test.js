import '@testing-library/jest-dom';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { combineReducers, createStore, applyMiddleware } from 'redux';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';
import { act } from 'react-dom/test-utils';
import App from '../App';
import leaguesReducer, { fetchLeaguesAction } from '../redux/teams/teams_reducer';

const reducer = combineReducers({
  leagues: leaguesReducer,
});

const store = createStore(
  reducer,
  applyMiddleware(thunk),
);

const mockApiData = [{
  id: '423e4faa-4c4b-4cf2-b1e9-03e8e5cace1f',
  name: 'Premier League',
  country: 'England',
  logo: 'https://media.api-sports.io/football/leagues/39.png',
  teams: [
    {
      id: '0b9d69df-5a2d-44c4-8502-3bbf32e6bb3a',
      name: 'Manchester United',
      city: 'Manchester',
      founded: 1878,
      details: 'Manchester United Football Club, commonly referred to as Man United, is a professional football club based in the Old Trafford area of Manchester, England. The club competes in the Premier League, the top division in the English football league system.',
      logo: 'https://media.api-sports.io/football/teams/33.png',
    },
    {
      id: '6ac5a9fd-9c5f-4b5e-b9a5-aeb3618890d6',
      name: 'Newcastle',
      city: 'Newcastle upon Tyne',
      founded: 1892,
      details: 'Newcastle United Football Club is an English professional football club based in Newcastle upon Tyne, that plays in the Premier League â€“ the top flight of English football. The club was founded in 1892 by the merger of Newcastle East End and Newcastle West End.',
      logo: 'https://media.api-sports.io/football/teams/34.png',
    }],
}];

const initialState = { leagues: [] };

const mockLoadLeagues = () => (fetchLeaguesAction(mockApiData));

describe('page render test:', () => {
  it('Homepage should be rendered correctly', () => {
    const page = renderer.create(
      <Provider store={store}>
          <App />
      </Provider>,
    ).toJSON();
    expect(page).toMatchSnapshot();
  });
});

describe('Redux store tests:', () => {
  it('Reducer returns initial state', () => {
    expect(store.getState()).toEqual(initialState);
  });

  it('fetchLeagesAction should be dispatched correctly', () => {
    act(() => {
      store.dispatch(mockLoadLeagues());
    });
    expect(store.getState().leagues).toEqual(mockApiData);
  });
});
