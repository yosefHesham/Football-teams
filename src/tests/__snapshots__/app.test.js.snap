// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page render test: Homepage renders correctly 1`] = `
<main>
  <nav
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#161a1d",
        "display": "flex",
        "height": "10vh",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <h1
      style={
        Object {
          "margin": "0 auto",
        }
      }
    >
      Football Leagues
    </h1>
    <ul
      style={
        Object {
          "columnGap": "15px",
          "display": "flex",
          "listStyle": "none",
          "marginRight": "10px",
        }
      }
    >
      <li>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-microphone c-white"
          data-icon="microphone"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 352c53.03 0 96-42.97 96-96v-160c0-53.03-42.97-96-96-96s-96 42.97-96 96v160C96 309 138.1 352 192 352zM344 192C330.7 192 320 202.7 320 215.1V256c0 73.33-61.97 132.4-136.3 127.7c-66.08-4.169-119.7-66.59-119.7-132.8L64 215.1C64 202.7 53.25 192 40 192S16 202.7 16 215.1v32.15c0 89.66 63.97 169.6 152 181.7V464H128c-18.19 0-32.84 15.18-31.96 33.57C96.43 505.8 103.8 512 112 512h160c8.222 0 15.57-6.216 15.96-14.43C288.8 479.2 274.2 464 256 464h-40v-33.77C301.7 418.5 368 344.9 368 256V215.1C368 202.7 357.3 192 344 192z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </li>
      <li>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-gear c-white"
          data-icon="gear"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M495.9 166.6C499.2 175.2 496.4 184.9 489.6 191.2L446.3 230.6C447.4 238.9 448 247.4 448 256C448 264.6 447.4 273.1 446.3 281.4L489.6 320.8C496.4 327.1 499.2 336.8 495.9 345.4C491.5 357.3 486.2 368.8 480.2 379.7L475.5 387.8C468.9 398.8 461.5 409.2 453.4 419.1C447.4 426.2 437.7 428.7 428.9 425.9L373.2 408.1C359.8 418.4 344.1 427 329.2 433.6L316.7 490.7C314.7 499.7 307.7 506.1 298.5 508.5C284.7 510.8 270.5 512 255.1 512C241.5 512 227.3 510.8 213.5 508.5C204.3 506.1 197.3 499.7 195.3 490.7L182.8 433.6C167 427 152.2 418.4 138.8 408.1L83.14 425.9C74.3 428.7 64.55 426.2 58.63 419.1C50.52 409.2 43.12 398.8 36.52 387.8L31.84 379.7C25.77 368.8 20.49 357.3 16.06 345.4C12.82 336.8 15.55 327.1 22.41 320.8L65.67 281.4C64.57 273.1 64 264.6 64 256C64 247.4 64.57 238.9 65.67 230.6L22.41 191.2C15.55 184.9 12.82 175.3 16.06 166.6C20.49 154.7 25.78 143.2 31.84 132.3L36.51 124.2C43.12 113.2 50.52 102.8 58.63 92.95C64.55 85.8 74.3 83.32 83.14 86.14L138.8 103.9C152.2 93.56 167 84.96 182.8 78.43L195.3 21.33C197.3 12.25 204.3 5.04 213.5 3.51C227.3 1.201 241.5 0 256 0C270.5 0 284.7 1.201 298.5 3.51C307.7 5.04 314.7 12.25 316.7 21.33L329.2 78.43C344.1 84.96 359.8 93.56 373.2 103.9L428.9 86.14C437.7 83.32 447.4 85.8 453.4 92.95C461.5 102.8 468.9 113.2 475.5 124.2L480.2 132.3C486.2 143.2 491.5 154.7 495.9 166.6V166.6zM256 336C300.2 336 336 300.2 336 255.1C336 211.8 300.2 175.1 256 175.1C211.8 175.1 176 211.8 176 255.1C176 300.2 211.8 336 256 336z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </li>
    </ul>
  </nav>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-spinner "
    data-icon="spinner"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M304 48C304 74.51 282.5 96 256 96C229.5 96 208 74.51 208 48C208 21.49 229.5 0 256 0C282.5 0 304 21.49 304 48zM304 464C304 490.5 282.5 512 256 512C229.5 512 208 490.5 208 464C208 437.5 229.5 416 256 416C282.5 416 304 437.5 304 464zM0 256C0 229.5 21.49 208 48 208C74.51 208 96 229.5 96 256C96 282.5 74.51 304 48 304C21.49 304 0 282.5 0 256zM512 256C512 282.5 490.5 304 464 304C437.5 304 416 282.5 416 256C416 229.5 437.5 208 464 208C490.5 208 512 229.5 512 256zM74.98 437C56.23 418.3 56.23 387.9 74.98 369.1C93.73 350.4 124.1 350.4 142.9 369.1C161.6 387.9 161.6 418.3 142.9 437C124.1 455.8 93.73 455.8 74.98 437V437zM142.9 142.9C124.1 161.6 93.73 161.6 74.98 142.9C56.24 124.1 56.24 93.73 74.98 74.98C93.73 56.23 124.1 56.23 142.9 74.98C161.6 93.73 161.6 124.1 142.9 142.9zM369.1 369.1C387.9 350.4 418.3 350.4 437 369.1C455.8 387.9 455.8 418.3 437 437C418.3 455.8 387.9 455.8 369.1 437C350.4 418.3 350.4 387.9 369.1 369.1V369.1z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</main>
`;

exports[`page render test: Homepage should be rendered correctly 1`] = `
<main>
  <nav
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#161a1d",
        "display": "flex",
        "height": "10vh",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <h1
      style={
        Object {
          "margin": "0 auto",
        }
      }
    >
      Football Leagues
    </h1>
    <ul
      style={
        Object {
          "columnGap": "15px",
          "display": "flex",
          "listStyle": "none",
          "marginRight": "10px",
        }
      }
    >
      <li>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-microphone c-white"
          data-icon="microphone"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 352c53.03 0 96-42.97 96-96v-160c0-53.03-42.97-96-96-96s-96 42.97-96 96v160C96 309 138.1 352 192 352zM344 192C330.7 192 320 202.7 320 215.1V256c0 73.33-61.97 132.4-136.3 127.7c-66.08-4.169-119.7-66.59-119.7-132.8L64 215.1C64 202.7 53.25 192 40 192S16 202.7 16 215.1v32.15c0 89.66 63.97 169.6 152 181.7V464H128c-18.19 0-32.84 15.18-31.96 33.57C96.43 505.8 103.8 512 112 512h160c8.222 0 15.57-6.216 15.96-14.43C288.8 479.2 274.2 464 256 464h-40v-33.77C301.7 418.5 368 344.9 368 256V215.1C368 202.7 357.3 192 344 192z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </li>
      <li>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-gear c-white"
          data-icon="gear"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M495.9 166.6C499.2 175.2 496.4 184.9 489.6 191.2L446.3 230.6C447.4 238.9 448 247.4 448 256C448 264.6 447.4 273.1 446.3 281.4L489.6 320.8C496.4 327.1 499.2 336.8 495.9 345.4C491.5 357.3 486.2 368.8 480.2 379.7L475.5 387.8C468.9 398.8 461.5 409.2 453.4 419.1C447.4 426.2 437.7 428.7 428.9 425.9L373.2 408.1C359.8 418.4 344.1 427 329.2 433.6L316.7 490.7C314.7 499.7 307.7 506.1 298.5 508.5C284.7 510.8 270.5 512 255.1 512C241.5 512 227.3 510.8 213.5 508.5C204.3 506.1 197.3 499.7 195.3 490.7L182.8 433.6C167 427 152.2 418.4 138.8 408.1L83.14 425.9C74.3 428.7 64.55 426.2 58.63 419.1C50.52 409.2 43.12 398.8 36.52 387.8L31.84 379.7C25.77 368.8 20.49 357.3 16.06 345.4C12.82 336.8 15.55 327.1 22.41 320.8L65.67 281.4C64.57 273.1 64 264.6 64 256C64 247.4 64.57 238.9 65.67 230.6L22.41 191.2C15.55 184.9 12.82 175.3 16.06 166.6C20.49 154.7 25.78 143.2 31.84 132.3L36.51 124.2C43.12 113.2 50.52 102.8 58.63 92.95C64.55 85.8 74.3 83.32 83.14 86.14L138.8 103.9C152.2 93.56 167 84.96 182.8 78.43L195.3 21.33C197.3 12.25 204.3 5.04 213.5 3.51C227.3 1.201 241.5 0 256 0C270.5 0 284.7 1.201 298.5 3.51C307.7 5.04 314.7 12.25 316.7 21.33L329.2 78.43C344.1 84.96 359.8 93.56 373.2 103.9L428.9 86.14C437.7 83.32 447.4 85.8 453.4 92.95C461.5 102.8 468.9 113.2 475.5 124.2L480.2 132.3C486.2 143.2 491.5 154.7 495.9 166.6V166.6zM256 336C300.2 336 336 300.2 336 255.1C336 211.8 300.2 175.1 256 175.1C211.8 175.1 176 211.8 176 255.1C176 300.2 211.8 336 256 336z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </li>
    </ul>
  </nav>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-spinner "
    data-icon="spinner"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M304 48C304 74.51 282.5 96 256 96C229.5 96 208 74.51 208 48C208 21.49 229.5 0 256 0C282.5 0 304 21.49 304 48zM304 464C304 490.5 282.5 512 256 512C229.5 512 208 490.5 208 464C208 437.5 229.5 416 256 416C282.5 416 304 437.5 304 464zM0 256C0 229.5 21.49 208 48 208C74.51 208 96 229.5 96 256C96 282.5 74.51 304 48 304C21.49 304 0 282.5 0 256zM512 256C512 282.5 490.5 304 464 304C437.5 304 416 282.5 416 256C416 229.5 437.5 208 464 208C490.5 208 512 229.5 512 256zM74.98 437C56.23 418.3 56.23 387.9 74.98 369.1C93.73 350.4 124.1 350.4 142.9 369.1C161.6 387.9 161.6 418.3 142.9 437C124.1 455.8 93.73 455.8 74.98 437V437zM142.9 142.9C124.1 161.6 93.73 161.6 74.98 142.9C56.24 124.1 56.24 93.73 74.98 74.98C93.73 56.23 124.1 56.23 142.9 74.98C161.6 93.73 161.6 124.1 142.9 142.9zM369.1 369.1C387.9 350.4 418.3 350.4 437 369.1C455.8 387.9 455.8 418.3 437 437C418.3 455.8 387.9 455.8 369.1 437C350.4 418.3 350.4 387.9 369.1 369.1V369.1z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</main>
`;
